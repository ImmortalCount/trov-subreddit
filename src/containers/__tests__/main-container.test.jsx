import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import MainContainer from '../main-container'

describe('MainContainer', () => {
    const wrapper = shallow(<MainContainer />)

    it('should render MainContainer', () => {
        expect(wrapper.find('Router').length).toBe(1)
        expect(wrapper.find('Route').length).toBe(3)
    })

    it('MainContainer matches snapshot', () => {
        const mainContainerRender = renderer.create(
            <MainContainer />
        ).toJSON()
        expect(mainContainerRender).toMatchSnapshot()
    })
})
