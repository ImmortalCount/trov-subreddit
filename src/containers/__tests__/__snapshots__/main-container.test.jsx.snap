// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer MainContainer matches snapshot 1`] = `
<div
  className="sc-bdVaJa enHcpj"
>
  <header
    className="sc-ifAKCX gcPEBf"
  >
    <a
      className="sc-bxivhb drpxUl"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <fieldset
      className="sc-htpNat PiYSI"
    >
      <label
        htmlFor="subreddit-select"
      >
        Select a Suggested Subreddit
      </label>
      <select
        className="sc-EHOje jvXsky"
        id="subreddit-select"
        onChange={[Function]}
      >
        <optgroup
          label="Select a Suggested Subreddit"
        >
          <option
            value="reactjs"
          >
            reactjs
          </option>
          <option
            value="guitarpedals"
          >
            Guitar Pedals
          </option>
          <option
            value="guitar"
          >
            Guitar
          </option>
          <option
            value="dataisbeautiful"
          >
            dataisbeautiful
          </option>
          <option
            value="DIY"
          >
            DIY
          </option>
          <option
            value="doommetal"
          >
            Doom Metal
          </option>
          <option
            value="javascript"
          >
            javascript
          </option>
          <option
            value="nottheonion"
          >
            Not The Onion
          </option>
        </optgroup>
      </select>
    </fieldset>
    <form
      className="sc-bZQynM epBIuB"
      onSubmit={[Function]}
    >
      <input
        name="subreddit"
        placeholder="Enter a subreddit..."
      />
      <button
        className="sc-bwzfXH cEwmJl"
        color="green"
        onClick={[Function]}
        size="small"
      >
        submit
      </button>
    </form>
  </header>
</div>
`;
